src/consumption.cpp

```diff
@@ -561,6 +569,9 @@ int Character::vitamin_mod( const vitamin_id &vit, int qty )
         it->second = std::min( it->second + qty, v.max() );
         update_vitamins( vit );
 
+        // update the daily trackers too while here
+        daily_vitamins[vit].second += qty;
```

Comment:
mac ubsan tests are failing here
```
(~[slow] ~[.],starting_items)=> src/consumption.cpp:573:36: runtime error: signed integer overflow: 2147483647 + 2147481247 cannot be represented in type 'int'
```

Reply:
Tests caught it too.
https://github.com/CleverRaven/Cataclysm-DDA/runs/7663399494?check_suite_focus=true#step:16:329

Reply:
To be fixed in #59952, looks like